inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -0,0 +1,374 @@
+    if (vec.empty()) return "";
+    std::stringstream oss;
+    oss << vec[0];
+    for (size_t i = 1; i < vec.size(); i++) oss << glue << vec[i];
+    return oss.str();
+}
+}  // namespace
+
+namespace {  // implementation details
+struct Edge {
+    int from_layer = 0;
+    int from_port = 0;
+    int to_layer = 0;
+    int to_port = 0;
+};
+
+struct ConstantAtributes {
+    int size = 0;
+    int offset = 0;
+};
+
+class XmlVisitor : public ngraph::AttributeVisitor {
+    pugi::xml_node m_data;
+
+    template <typename T>
+    std::string create_atribute_list(
+        ngraph::ValueAccessor<std::vector<T>>& adapter) {
+        return joinVec(adapter.get(), std::string(","));
+    }
+
+public:
+    XmlVisitor(pugi::xml_node& data) : m_data(data) {}
+
+    void on_adapter(const std::string& name,
+                    ngraph::ValueAccessor<void>& adapter) override {
+#if 0  // TODO: remove when Constant will support VisitorAPI
+        m_data.append_attribute(name.c_str());
+#endif
+    }
+    void on_adapter(const std::string& name,
+                    ngraph::ValueAccessor<bool>& adapter) override {
+        m_data.append_attribute(name.c_str()).set_value(adapter.get());
+    }
+    void on_adapter(const std::string& name,
+                    ngraph::ValueAccessor<std::string>& adapter) override {
+        m_data.append_attribute(name.c_str()).set_value(adapter.get().c_str());
+    }
+    void on_adapter(const std::string& name,
+                    ngraph::ValueAccessor<int64_t>& adapter) override {
+        m_data.append_attribute(name.c_str()).set_value(adapter.get());
+    }
+    void on_adapter(const std::string& name,
+                    ngraph::ValueAccessor<double>& adapter) override {
+        m_data.append_attribute(name.c_str()).set_value(adapter.get());
+    }
+    void on_adapter(
+        const std::string& name,
+        ngraph::ValueAccessor<std::vector<int64_t>>& adapter) override {
+        m_data.append_attribute(name.c_str())
+            .set_value(create_atribute_list(adapter).c_str());
+    }
+    void on_adapter(
+        const std::string& name,
+        ngraph::ValueAccessor<std::vector<uint64_t>>& adapter) override {
+        m_data.append_attribute(name.c_str())
+            .set_value(create_atribute_list(adapter).c_str());
+    }
+    void on_adapter(
+        const std::string& name,
+        ngraph::ValueAccessor<std::vector<float>>& adapter) override {
+        m_data.append_attribute(name.c_str())
+            .set_value(create_atribute_list(adapter).c_str());
+    }
+    void on_adapter(
+        const std::string& name,
+        ngraph::ValueAccessor<std::vector<std::string>>& adapter) override {
+        m_data.append_attribute(name.c_str())
+            .set_value(create_atribute_list(adapter).c_str());
+    }
+};
+
+const std::map<ngraph::Node*, int> create_layer_ids(const ngraph::Function& f) {
+    std::map<ngraph::Node*, int> layer_ids;
+    int id = 0;
+    for (auto node : f.get_ordered_ops()) {
```

Comment:
```suggestion
    for (const auto& node : f.get_ordered_ops()) {
```

Reply:
Done in this [commit](https://github.com/openvinotoolkit/openvino/pull/2579/commits/09fbad0067999b153c2c78be728aafef935d25aa)